src/main/java/org/verdictdb/coordinator/ExecutionContext.java

```diff
@@ -144,10 +144,17 @@ public VerdictSingleResult sql(String query, boolean getResult) throws VerdictDB
       if (stream == null) {
         return null;
       }
-      AsyncAggExecutionRulerByCount ruler = new AsyncAggExecutionRulerByCount(1, stream, runningCoordinator);
+      QueryResultAccuracyEstimator accEst = new QueryResultAccuracyEstimatorWithGroupSize(1, stream, runningCoordinator);
       try {
-        VerdictSingleResult result = ruler.fetchAnswerUntilAccurate();
-        return result;
+        while (stream.hasNext()) {
+          VerdictSingleResult rs = stream.next();
+          accEst.add(rs);
+          if (accEst.isLastResultAccurate()) {
+            return rs; // also do abort the query execution
+          }
+        }
+        // return the last result otherwise
+        return accEst.getAnswers().get(accEst.answers.size()-1);
```

Comment:
Instead of using `accEst.answers.size()`, let's create a method `accEst.getAnswerCount()` and use it. This is simply my personal prefernece.